packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx

```diff
@@ -22,13 +37,40 @@ export const DashboardExtension = ({
   const location = useLocation();
   const navLink = useMemo(() => decodeURIComponent(last(location.pathname.split('/'))), [location.pathname]);
 
+  type MenuTitle = keyof typeof MenuIcons;
+
+  const MenuIcons = {
+    'Patient Summary': ReportIcon,
+    'Vitals & Biometrics': ActivityIcon,
+    Medications: MedicationIcon,
+    Orders: ShoppingCartIcon,
+    Results: ChartAverageIcon,
+    Visits: CalendarHeatMapIcon,
+    Allergies: WarningIcon,
+    Conditions: ListCheckedIcon,
+    Immunizations: SyringeIcon,
+    Attachments: DocumentAttachmentIcon,
+    Programs: ProgramsIcon,
+    Appointments: EventScheduleIcon,
+  } as const;
+
+  const menuIcon = (title: MenuTitle) => {
+    const Icon = MenuIcons[title];
+    return Icon ? <Icon className={styles.icon} /> : null;
+  };
+
+  const renderIcon = menuIcon(title as MenuTitle);
```

Comment:
No weird edge-cases and we don't have to lie to TypeScript:

```suggestion
  const renderIcon = Object.keys(MenuIcons).includes(title) ? menuIcon(title) : null;
```